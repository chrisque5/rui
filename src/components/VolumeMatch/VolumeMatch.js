import PropTypes from 'prop-types';
import { Form, Checkbox } from 'antd';

const VolumeMatch = ({
  id, label,
}) => (
  <Form.Item className="volume-match" valuePropName="checked" name={id}>
    <Checkbox>{label}</Checkbox>
  </Form.Item>
);

VolumeMatch.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
};

export default VolumeMatch;
