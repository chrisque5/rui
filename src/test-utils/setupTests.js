// The below part of the code is to fix the issue
// TypeError: window.matchMedia is not a function
Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});
