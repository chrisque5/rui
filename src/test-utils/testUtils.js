/* eslint-disable import/no-extraneous-dependencies, react/prop-types */
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Form } from 'antd';

const middlewares = [thunk];
const renderWithForm = (
  ui,
  {
    initialState,
    store = configureMockStore(middlewares)(initialState),
    ...renderOptions
  } = {},
) => {
  const wrapper = ({ children }) => <Provider store={store}><Form>{children}</Form></Provider>;

  return render(ui, { wrapper, ...renderOptions });
};

const renderWithProvider = (
  ui,
  {
    initialState,
    store = configureMockStore(middlewares)(initialState),
    ...renderOptions
  } = {},
) => {
  const wrapper = ({ children }) => <Provider store={store}>{children}</Provider>;

  return render(ui, { wrapper, ...renderOptions });
};
// re-export everything
export * from '@testing-library/react';
// override render method
export { renderWithForm, renderWithProvider };
