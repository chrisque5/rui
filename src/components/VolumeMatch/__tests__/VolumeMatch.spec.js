import { renderWithForm, fireEvent, waitFor } from 'test-utils/testUtils';
import VolumeMatch from '../VolumeMatch';

const mockForm = {};
const defaultProps = { form: mockForm, id: 'volumeMatch', label: 'Volume Match' };

describe('<VolumeMatch />', () => {
  it('should render snapshot as expected', () => {
    const { container } = renderWithForm(<VolumeMatch {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it('should handle clicking of the checkbox', async () => {
    const { getByLabelText } = renderWithForm(<VolumeMatch {...defaultProps} />);
    const checkbox = await waitFor(() => getByLabelText('Volume Match'));
    expect(checkbox.checked).toEqual(false);

    fireEvent.click(checkbox);
    expect(checkbox.checked).toEqual(true);

    fireEvent.click(checkbox);
    expect(checkbox.checked).toEqual(false);
  });
});
