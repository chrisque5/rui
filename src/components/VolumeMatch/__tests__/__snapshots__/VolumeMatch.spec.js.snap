// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VolumeMatch /> should render snapshot as expected 1`] = `
<div>
  <form
    class="ant-form ant-form-horizontal"
  >
    <div
      class="ant-row ant-form-item volume-match"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <label
              class="ant-checkbox-wrapper"
            >
              <span
                class="ant-checkbox"
              >
                <input
                  class="ant-checkbox-input"
                  id="volumeMatch"
                  type="checkbox"
                  value=""
                />
                <span
                  class="ant-checkbox-inner"
                />
              </span>
              <span>
                Volume Match
              </span>
            </label>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
